frontend/src/metabase/public/components/widgets/EmbedModalContent.unit.spec.tsx

```diff
@@ -0,0 +1,208 @@
+import React from "react";
+import { screen, waitFor } from "@testing-library/react";
+import _ from "underscore";
+import { renderWithProviders } from "__support__/ui";
+import EmbedModalContent from "./EmbedModalContent";
+
+const storeInitialState = {
+  currentUser: {
+    is_superuser: true,
+  },
+  settings: {
+    values: {
+      "enable-embedding": true,
+      "embedding-secret-key": "my_super_secret_key",
+    },
+  },
+};
+
+describe("EmbedModalContent", () => {
+  it("should render", () => {
+    renderWithProviders(
+      <EmbedModalContent
+        resource={{}}
+        resourceParameters={[]}
+        getPublicUrl={jest.fn()}
+      />,
+      {
+        storeInitialState,
+        reducers: {
+          settings: (
+            state: Record<string, any> = storeInitialState.settings,
+          ) => {
+            return state;
+          },
+        },
+      },
+    );
```

Comment:
WDYT about extracting a shared `setup` function or something that'd render the component and configure redux? I see identical steps duplicated across all the tests, so it's a bit hard to parse what exactly is different in each test

Reply:
Also, this test doesn't have any assertions

Reply:
Ahh yep, I forgot to extract those setup step thanks for calling that out.

Re: don't have any assertion. I'm not sure where I've seen this before but at the end of the day I just want to make sure we are able to render the component. But I think that we could still definitely assert certain elements that it's been render and that should be more explicit. 👍

Reply:
Fixed.
1. Extract repeating setup to `renderWithConfiguredProviders`, I don't like `setup` since it's too broad and doesn't sound like it would try to render anything imo.
2. Add some assertions to the first test.